### ISSUE TITLE
EmbeddedMap is broken

### ISSUE THREAD

**Issue by maseev** [orientechnologies/orientdb#5473]
I was trying to save and then query an entity which contains an embedded map (```Map<String, Car>``` to be precise). Here's a model that represents the database structure - https://github.com/maseev/orient/tree/master/src/main/java/orientdb/model. And here's the test where I'm trying to query an entity that was previously saved to the database - https://github.com/maseev/orient/blob/master/src/test/java/orientdb/OrientDbTest.java#L46.

The problem I faced to is instead of the entity that I save to the database I get the entity with an embedded map, where all values of the HashMap are equal to the first Car object that I put to the map. So, placeToCar map contains:
```
 "USA" -> Car("Cadillac Escalade", 1990)
 "Japan" -> Car("Cadillac Escalade", 1990)
 "UK" -> Car("Cadillac Escalade", 1990)
```
instead of:
```
 "USA" -> Car("Cadillac Escalade", 1990)
 "Japan" -> Car("Nissan Skyline", 2001)
 "UK" -> Car("Jaguar XR", 2007)
```
You can run ``mvn test`` to run the test. I clearly remember that way back when I was using OrientDb (it was 1.7 version) this functionality worked perfectly.

**Comment by maseev** [comment]
This behaviour is reproducible since orientdb 2.0.4.

**Comment by tglman** [comment]
@maseev 

Thank you for the detailed test, i could reproduce de problem, working on a fix.

bye

**Comment by maseev** [comment]
By the way, it would be great to have some sort of integration tests which would cover this use case in order to avoid regression in subsequent releases.

**Comment by maseev** [comment]
I'm just wondering, if this fix will go to the next hotfix release?

**Comment by tglman** [comment]
will be fixed in 2.1.9 and 2.2

**Comment by tglman** [comment]
2.1.9 was released closing and targeting
